### ISSUE TITLE
Not working on TYPO3 12?

### ISSUE THREAD

**Issue by boosim** [franzholz/tt_board#33]
Hello,

I've tried to install tt_board on TYPO3 12 exactly according to the documentation but unfortunately I was not able to get it to work.
No matter what I configure, there is no output by the extension.

The page gets rendered like usual and there is no error message. So it doesn't seem to be a PHP error or something like that. There is just nothing rendered by the extension.

Has anyone else tried to set it up in TYPO3 12 or are there any hints what I might have missed?

Thanks and kind regards
Florian

**Comment by franzholz** [comment]
The documentation is not rendered on extensionstypo3.org. You must look into the extension Documentation folder to get the real documenation.

**Comment by franzholz** [comment]
Use the extension debug_mysql_query to check which SQL queries are generated.

**Comment by franzholz** [comment]
You have not inserted the 'include static template from extensions' tt_ board static template in the Typoscript module in the backend.

**Comment by boosim** [comment]
Dear Franz,

thank you very much for your replies to this.

We used the tutorial that comes with the extension in doc/manual.odt
Is that the right one?

The static template for tt_board is included in the main template and the template in the data folder which is also included in the main template.

The extension debug_mysql_db did unfortunately show no queries in the Frontend, only in the Backend. I've activated Debug ode as well but still no results

Thans an kind regards
Florian

**Comment by franzholz** [comment]
No, the doc/manual.odt is the old documentation.

Try to debug the extension where it does not continue.
tt_board/Classes/Controller/RegisterPluginController.php:

`public function main($content, $conf)
`
Try to find out if this PHP code is executed .

`return $content
`
This is the output on the screen.

**Comment by boosim** [comment]
Thanks again for your reply.

Where can I find the current documentation?

Your Github page on https://github.com/franzholz/tt_board still states:
_The full documentation is still only in the file doc/manual.odt !_
That's why I thought I should look at that file instead of the one in the repository.

I have put first  **echo "TEST";** and then **return "TEST";** at the beginning of the above mentioned main function. Both don't appear anywhere on the website afterwards so it seems like the function is not executed.

There doesn't seem to be any error messages in the backend log or in typo3temp/var/logs as well.

**Comment by franzholz** [comment]
As I have written before, the documentation is in the Documentation folder of tt_board. 

If the PHP code of tt_board is not executed, then the Default Setup has not been inserted into your TypoScript. Check your TypoScript. 
You can also insert it manually:  

_tt_board/Configuration/TypoScript/Default/ ..._

**Comment by franzholz** [comment]
There are 2 possible plugins.
The other is

    temp.userFuncList = JambageCom\TtBoard\Controller\ListPluginController->main


plugin.tt_board_list {
  userFunc = JambageCom\TtBoard\Controller\ListPluginController->main
 }
 
 This must be visible in your Typoscript. 
 
 Then check by means of error_log or debug if the extension's code is called. 
 Use the Source Code of the HTML to check if you find any output  of a 'tt_board' Plugin.

**Comment by franzholz** [comment]
Try to add the Content Wizard in the TSConfig of the top page:

Page TSconfig
**Include static Page TSconfig (from extensions) [tsconfig_includes]** 


Check if it works after using the content wizard. Check if another extension plugin works.

**Comment by boosim** [comment]
Both temp.userFuncList and plugin.tt_board_list.userFunc are visible in the Active Typoscript.
The same is true for  temp.userFuncTree and plugin.tt_board_tree.userFunc.

There is no entry in any error log, whether I run in Debug or Live mode.

In the html code, the div container with the content ID is visible but inside it is completely empty.

I have cleared the cache more than once, including typo3temp/var/cache and autoload.


Including the TSconfig on the pages unfortunately didn't help.

There is at least one other Frontend Plugin extension in the web that works fine.

Could it be an issues with the PHP version? We are using PHP 8.2 for this account.


I will try it next week on an empty web to make sure there is no weird problem with another extension or configuration.

Thanks again and have a nice weekend
Florian

**Comment by franzholz** [comment]
No, this is no issue with PHP nor with tt_board. Otherwise you would have had an error_log output. But the PHP code of tt_board is never executed. 

You must also have this TypoScript:

```
tt_content.list.20.2 = CASE
tt_content.list.20.2 {
    key.field = layout
    0 = < plugin.tt_board_tree
}

tt_content.list.20.4 = CASE
tt_content.list.20.4 {
    key.field = layout
    0 = < plugin.tt_board_list
    1 = < plugin.tt_board_tree
}

```

**Comment by franzholz** [comment]
The documentation is also online:

https://docs.typo3.org/typo3cms/extensions/tt_board/stable/Introduction/Index.html

There is almost no difference to the current behaviour.

**Comment by typo3max** [comment]
There is really a post error in tt_board. with TYPO3 12

**Comment by franzholz** [comment]
What does this mean?

**Comment by typo3max** [comment]
You cant submit Forum Massages!

**Comment by typo3max** [comment]
Bitte mal kurz im Forum gucken bei mir da ist noch ein kleiner Fehler!
________________________________
Von: Franz Holzinger ***@***.***>
Gesendet: Donnerstag, 4. April 2024 13:01
An: franzholz/tt_board ***@***.***>
Cc: Maximilian Fasel ***@***.***>; Comment ***@***.***>
Betreff: Re: [franzholz/tt_board] Not working on TYPO3 12? (Issue #33)


Constant:

emailCheck = 0 under plugin.tt_board

This is even the default setting.

**Comment by franzholz** [comment]
Do not add additional questions here!

**Comment by franzholz** [comment]
The documentation is already rendered in the typo3 documentation page at

https://docs.typo3.org/p/jambagecom/tt-board/main/en-us/

**Comment by franzholz** [comment]
Could the problem be solved now? Give me a feedback!

**Comment by typo3max** [comment]
I have no problems anymore with forum
________________________________
Von: Franz Holzinger ***@***.***>
Gesendet: Montag, 6. Mai 2024 00:16
An: franzholz/tt_board ***@***.***>
Cc: Maximilian Fasel ***@***.***>; Comment ***@***.***>
Betreff: Re: [franzholz/tt_board] Not working on TYPO3 12? (Issue #33)


Could the problem be solved now? Give me a feedback!

**Comment by boosim** [comment]
Thanks again for your help. Unfortunately I did not get it to work in my project but I suspect there was some issue in combination with other extensions. 

I noticed that the extension "container" did not work correctly any more after installing tt_board. The display and functionality of that extension in the BE did not work completely any more. After deinstalling tt_board, the container extension did work again.

I tried to deinstall container and install tt_board but it still did not work. I wasn't able to investigate further at the moment. I will come back to you if I can find any more information on what the problem could be.

Kind regards
Florian
